
We're using a different reference clock than the downstream driver, and this clock changes with mode sets.

So every mode would have slightly different timings, almost all of them wrong. Fix this!

Signed-off: derek.foreman@collabora.com


diff --speed-large-files --no-dereference --minimal -Naur linux-6.12.2/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c linux-6.12.2/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
--- linux-6.12.2/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c	2024-12-05 21:27:34.581671865 +0100
+++ linux-6.12.2/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c	2024-12-05 21:07:56.218334472 +0100
@@ -1238,6 +1238,7 @@
 	struct drm_connector *connector;
 	struct drm_connector_state *conn_state;
 	void* data = hdmi->phy.data;
+	int rate;
 
 	connector = drm_atomic_get_new_connector_for_encoder(state,
 							     bridge->encoder);
@@ -1260,6 +1261,8 @@
 		mutex_unlock(&hdmi->audio_mutex);
 	}
 
+	rate = conn_state->crtc->state->adjusted_mode.crtc_clock * 1000;
+	dw_hdmi_qp_write(hdmi, rate, TIMER_BASE_CONFIG0);
 	dw_hdmi_qp_init_audio_infoframe(hdmi);
 	dw_hdmi_qp_audio_enable(hdmi);
 }
